src/ReadingPages.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Recipe } from "./recipe";
+import { Image } from "antd";
+import "antd/dist/antd.min.css";
+import "./Reading.css";
+
+export default function ReadingPage() {
+  const emptyRecipe: Recipe = {
+    id: 0,
+    name: "0",
+    thumbnail: {
+      image: "",
+    },
+    description: "",
+    link: ["0"],
+    metadata: {
+      lastViewed: "0",
+      created: "0",
+      timeToCook: "0",
+    },
+    ingredients: [
+      {
+        ingredient: "",
+        starred: true,
+      },
+    ],
+    steps: [
+      {
+        step: 0,
+        directions: "",
+        image: "",
+      },
+    ],
+  };
+
+  const [readingRecipe, setReadingRecipe] = useState<Recipe>(emptyRecipe);
+  const [loading, setLoading] = useState(true);
+
+  const id = useParams().recipeId;
```

Comment:
I would move this fetching logic to a separate file, that way you dont need to have it 3 times